presto-main/src/main/java/com/facebook/presto/operator/project/PageProcessor.java

```diff
@@ -162,10 +183,21 @@ protected Page computeNext()
                     }
                 }
 
+                updateRetainedSize();
                 return page;
             }
         }
 
+        private void updateRetainedSize()
+        {
+            retainedSizeInBytes = calculateRetainedSizeWithoutLoading(page);
+            for (Block previouslyComputedResult : previouslyComputedResults) {
+                if (previouslyComputedResult != null) {
+                    retainedSizeInBytes += previouslyComputedResult.getRetainedSizeInBytes();
```

Comment:
Sorry to jump in a little bit late; What if the page is loaded and the blocks in `previouslyComputedResult` are exactly the same (in terms of identity) as those in the `page`; will we double count the size?